redis/cache/redisx/redisxtest/redisxtest.go

```diff
@@ -0,0 +1,39 @@
+package redisxtest
+
+import (
+	"context"
+	"testing"
+
+	"github.com/joomcode/redispipe/redis"
+	"github.com/joomcode/redispipe/redisconn"
+
+	"github.com/reddit/baseplate.go/redis/cache/redisx"
+)
+
+// NewMockRedisClient sets up a redis client for use in testing
+func NewMockRedisClient(
+	ctx context.Context,
+	tb testing.TB,
+	address string,
+	opts redisconn.Opts,
+) (*redisx.BaseSync, error) {
```

Comment:
looking at how existing tests work, we do need the returned client to be of type `redisx.BaseSync` instead of the pointer, for example https://github.com/reddit/baseplate.go/blob/a6f0c8aa515567b05b5da3820b3aa879d33a1ce4/redis/cache/redispipebp/errors_test.go#L57